scripts/timing.py

```diff
@@ -40,27 +53,88 @@ def main():
         make_case_tilt_up_once(),
         make_case_tilt_up_again_no_change(),
         make_case_tilt_up_again_new_file(),
+        make_case_watch(),
     ]
     results = []
 
     try:
         for c in cases:
-            print('~~ RUNNING CASE: %s' % c.name)
-            c.setup()
-            timetake = c.test()
+            print('~~ RUNNING CASE: {}'.format(c.name))
+            args = []
+            kwargs = {}
+
+            print('~~~~ setup: {}'.format(c.name))
+            ret = c.setup()
+            if ret is not None:
+                args = ret[0]
```

Comment:
this API seems...weird.

specifically, the setup() and test() methods are tightly coupled, and using them correctly requires a lot of inside knowledge about how they're implemented.

possible alternatives:
1) is there a reason to have separate setup and test methods at all? i'm not sure that the separation has any upsides
2) make case a real class that keeps internal state

wdyt?

Reply:
ya, not blocking or anything